numba/npyufunc/ufuncbuilder.py

```diff
@@ -264,7 +264,11 @@ def __init__(self, py_func, signature, identity=None, cache=False,
         self.identity = parse_identity(identity)
         self.nb_func = jit(target='npyufunc', cache=cache)(py_func)
         self.signature = signature
-        self.sin, self.sout = parse_signature(signature)
+        # allow internal use to pass in data structure that represents signature
+        if isinstance(signature, tuple):
```

Comment:
The only place that this is used is in parfor.py . 
As commented in https://github.com/numba/numba/pull/2318/files#r109723364 , its usage is not needed; thus, this also seems unnecessary.